public/scripts/view-util.js

```diff
@@ -288,33 +288,62 @@ var ViewUtil = (function($) {
 	    '</div>';
     }
 
+    function splitHostWithPort(hostWithPort) {
+        // the yarn_job_history and jt_url variables (usually?) are not full URIs,
+        // so we cannot parse them directly as such.  This parsing approach
+        // here assumes hostWithPort is _not_ a URI, as it would fail otherwise.
+        // However this is consistent with the historical behavior of Sahale
+        var split = hostWithPort.split(':');
+
+        return { host: split[0], port: (split.length < 2) ? null : split[1] };
+    }
+
+    function buildHref(schemeWithSlashes, rawHost, port, path) {
+        // schemeWithSlashes is what it sounds like, e.g.
+        //  `http://` or `https://` (or `//`, ie inherit the current scheme)
```

Comment:
Do you think it would make sense to give `schemeWithSlashes` a default value of '//' so that if it's left off, the link will inherit the current schema? (Not a huge deal either way)

Reply:
hey sure good idea, that's a good default and that's what most of the existing calls do already so it will be nice to simplify them.